src/sidebar/services/analytics.ts

```diff
@@ -0,0 +1,20 @@
+import type { APIService } from './api';
+
+/**
+ * @inject
+ */
+export class AnalyticsService {
+  private _api: APIService;
+
+  constructor(api: APIService) {
+    this._api = api;
+  }
+
+  trackApplyPendingUpdatesEvent(): void {
```

Comment:
This API design will require us to add a new method for each event we want to track, which is going to be annoying. I suggest we just have one method that takes the event name.

Reply:
Yep, I thought about that, and agree.

I might even allow event objects to be passed, instead of event names, as we might want to add extra metadata, but perhaps we can delay that.

Using event classes might help there, so that we can do things like:

```ts
type AnalyticsEvent = {
  ...
}

class ApplyPendingUpdatesEvent implements AnalyticsEvent {
  event: 'APPLY_PENDING_UPDATES';
}

class SomeOtherEvent implements AnalyticsEvent {
  event: '...';
  moreStuff: {};
}

// ...

analyticsService.trackEvent(new ApplyPendingUpdatesEvent());
```

But as I said, let's start with event names for now and see how this evolves.